### PR TITLE
fix(Rect): fixes underflow in the `Rect::intersection` method

### PR DESCRIPTION
**Pull request by Valentin271**
From @bugnano in #677

> The `Rect.intersection` method contained the simple `-` operator, which panics on underflow.
> This is strange, as all the other methods in `Rect` use `saturating_add` and `saturating_sub`.
> So I changed the `-` operation to a `saturating_sub`, so that `Rect.intersection` never panics.

With a signed commit and a unit test.

### CODE REVIEW

**Review by orhun — APPROVED** [review]

**Comment by orhun** [comment]
Not sure if we need more unit tests for this, let's wait for review from other maintainers.

**Review by joshka — APPROVED** [review]
Seems like enough tests to me

**Pull request merged**
